x-pack/platform/plugins/shared/stack_alerts/server/rule_types/es_query/lib/fetch_es_query.ts

```diff
@@ -160,3 +164,39 @@ export async function fetchEsQuery({
     index: params.index,
   };
 }
+
+export function generateLink(
+  params: OnlyEsQueryRuleParams,
+  rawFilter: Record<string, unknown>,
+  discoverLocator: LocatorPublic<DiscoverAppLocatorParams>,
+  dateStart: string,
+  dateEnd: string,
+  spacePrefix: string
+) {
+  const dataViewId = 'es_query_rule_adhoc_data_view';
+  const filter = buildCustomFilter(
+    dataViewId,
+    rawFilter,
+    false,
+    false,
+    'Rule query DSL',
+    FilterStateStore.APP_STATE
+  );
+  const redirectUrlParams: DiscoverAppLocatorParams = {
+    dataViewSpec: {
+      id: dataViewId,
+      title: params.index.join(','),
+      timeFieldName: params.timeField,
+    },
+    filters: [filter],
+    timeRange: { from: dateStart, to: dateEnd },
+    isAlertResults: true,
+  };
+
+  // use `lzCompress` flag for making the link readable during debugging/testing
+  // const redirectUrl = discoverLocator!.getRedirectUrl(redirectUrlParams, { lzCompress: false });
+  const redirectUrl = discoverLocator!.getRedirectUrl(redirectUrlParams);
+  const [start, end] = redirectUrl.split('/app');
```

Comment:
Could this break if the path in their publicBaseURL has `/app` in it somewhere (for example, using some proxy in front of Kibana, like we do in dev).  Or if it's in a non-default space that starts with `app` (for example space "appXYZ" would have a URL like `host:port/s/appXYZ/app/...`

Reply:
Oh true, that might break something if there is `app` somewhere in the url. Do you think it's safer to split on `/app/`?